ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/model/TestCaseNode.java

```diff
@@ -133,12 +134,22 @@ public VersionBoundsNode versionBounds() {
   }
 
   private void validate() {
-    if (this.name.isEmpty()) {
+    if (name.isEmpty()) {
       throw new MissingFieldException("name");
     }
 
-    if (this.statements.isEmpty()) {
-      throw new InvalidFieldException("statements", "was empty");
+    try {
+      if (statements.isEmpty()) {
+        throw new InvalidFieldException("statements", "was empty");
+      }
+
+      if (!expectedException.isPresent() && inputs.isEmpty() && outputs.isEmpty()) {
```

Comment:
How come we only require one of inputs or outputs to be present? Under what scenario would it make sense to provide one and not the other?

Reply:
I now understand why the input file may be null: we might have tests that start with no inputs and instead use `INSERT VALUES` to insert data, but I still don't understand why it'd be useful to have a test where the output file is null. FWIW it looks like the testing tool requires the output file to be present: https://github.com/confluentinc/ksql/blob/c5d6b56a3188f100ea473ef8f13959da79d1c79e/ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/KsqlTestingTool.java#L55

Reply:
A test where a filter filters everything out.  Weird... yes... do we have such tests... yes.

Reply:
But shouldn't a test like that require an empty outputs node rather than a missing outputs node? Having a missing outputs node is counterintuitive since missing nodes in other parts of the test specification indicate that nothing should be checked, whereas in a test like that we would still want to verify that no outputs were produced.